src/Sentry/SpanContext.cs

```diff
@@ -37,17 +37,17 @@ public class SpanContext : ITraceContext
     /// Initializes an instance of <see cref="SpanContext"/>.
     /// </summary>
     public SpanContext(
-        SpanId spanId,
+        SpanId? spanId,
```

Comment:
Thanks for weeding out the constructors.
Since this is public and user-facing, can we make the final one even more user-friendly? If I were to create a new instance without propagating anything it would look like 
`var s = new SpanContext(null, null, null, "my-operation", null, null, null);`
Can we use defaults like with the new `TransactionContext` constructor?